src/ert/_c_wrappers/enkf/config/parameter_config.py

```diff
@@ -1,16 +1,29 @@
 from __future__ import annotations
 
+import dataclasses
```

Comment:
A general comment to `parameter_config.py` is that it is not a cwrapper and should be moved somewhere else.
No need to do that in this PR, so I have created an issue for it: https://github.com/equinor/ert/issues/5395
I also think that we should add type-hints even though cwrappers are not checked by mypy.